crates/pipeline_manager/src/db.rs

```diff
@@ -498,44 +521,56 @@ CREATE TABLE IF NOT EXISTS pipeline (
 
     /// Update project status.
     ///
-    /// Note: doesn't check that the project exists.
+    /// # Note
+    /// - Doesn't check that the project exists.
+    /// - Resets schema to null.
     fn set_project_status(&self, project_id: ProjectId, status: ProjectStatus) -> AnyResult<()> {
         let (status, error) = status.to_columns();
 
         self.dbclient
             .execute(
-                "UPDATE project SET status = $1, error = $2, status_since = unixepoch('now') WHERE id = $3",
+                "UPDATE project SET status = $1, error = $2, schema = null, status_since = unixepoch('now') WHERE id = $3",
                 (&status, &error, &project_id.0),
             )?;
 
         Ok(())
     }
 
-    /// Update project status after a version check.
+    /// Update project status and potentially the schema after a version check.
     ///
-    /// Updates project status to `status` if the current project version
-    /// in the database matches `expected_version`.
+    /// Updates project status to `status` if the current project version in the
+    /// database matches `expected_version`. Sets the schema if
+    /// `schema.is_some()`. Otherwise, the schema is not updated.
     ///
     /// # Note
     /// This intentionally does not throw an error if there is a project version
-    /// mismatch and instead does just not update. It's used by the compiler
-    /// to update status and in case there is a newer version it is expected
-    /// that the compiler just picks up and runs the next job.
+    /// mismatch and instead does just not update. It's used by the compiler to
+    /// update status and in case there is a newer version it is expected that
+    /// the compiler just picks up and runs the next job.
     pub(crate) fn set_project_status_guarded(
```

Comment:
One thing: maybe it's better to just do it with two queries in a transaction rather than polluting the set_project_status_guarded api.